docs(build): Moving developer setup to build.md (#444)
openebs/node-disk-manager: build.md

  * Build the binary.
  * Build the docker image with the binary.

> * To build multi-arch images, you need to have `buildx` installed. You can follow these [instructions](https://github.com/docker/buildx#installing).`

Reviewer: Typo. extra backtick
```suggestion
* To build multi-arch images, you need to have `buildx` installed. You can follow these [instructions](https://github.com/docker/buildx#installing).
```

  * NDM multi-arch images are built using QEMU mode. If you're on linux, run :
    * docker run --rm --privileged linuxkit/binfmt:v0.8

>   * Set the value of envirionment variable of `IMAGE_ORG` based on your user name at Dockerhub:  

Reviewer: ```suggestion
  * Set the value of envirionment variable of `IMAGE_ORG` based on your user name at Dockerhub:  This is required because `buildx` cannot store the multiarch image locally, instead pushes to the registry after building.
```

    * export IMAGE_ORG=name
  * Depending on what you're working on, run:
    * For node-disk-manager:
      * make docker.buildx.ndm
    * For node-disk-operator:
      * make docker.buildx.ndo
    * For node-disk-exporter:
      * make docker.buildx.exporter

> 

Reviewer: @harshthakur9030  Can you also provide a note that building multiarch image can take a lot of time since the build runs in a QEMU emulated environment

Reviewer: Also add to the note, the list of currently supported architectures: (amd64, arm64, armv7, ppc64le)

  * Build the binary.
  * Build the docker image with the binary.

> * To build multi-arch images, you need to have `buildx` installed. You can follow these [instructions](https://github.com/docker/buildx#installing).

Reviewer: ```suggestion
* To build multi-arch images, you need to have `buildx` installed. Follow these [instructions](https://github.com/docker/buildx#installing) to install buildx.
```

> Supported architectures are: amd64, arm64, armv7, ppc64le. 

Reviewer: Add a blank line. Supported architectures should come on a new line.

>   * NDM multi-arch images are built using QEMU mode(Note: This can take a lot of time since it runs in an emulated environment). If you're on linux, run :

Reviewer: ```suggestion
  * NDM multi-arch images are built using QEMU mode (**NOTE**: This can take a lot of time since it runs in an emulated environment). If you're on linux, run :
```

>     * docker run --rm --privileged linuxkit/binfmt:v0.8

Reviewer: should put the commands in a code block.

>   * Set the value of envirionment variable of `IMAGE_ORG` based on your user name at Dockerhub or any other registry which supports storing multi-arch images( This is required because multi-arch images cannot be stored locally so we push to the registry after building):  

Reviewer: ```suggestion
  * Set the value of environment variable of `IMAGE_ORG` based on your user name at Dockerhub or any other registry which supports storing multi-arch images (This is required because multi-arch images cannot be stored locally, so `buildx` pushes to the registry after building):  
```

>     * export IMAGE_ORG=name

Reviewer: Here also, put the command in code block.

  * Depending on what you're working on, run:
    * For node-disk-manager:

>       * make docker.buildx.ndm

Reviewer: These 3 make commands also should be in code block.

  * Build the binary.
  * Build the docker image with the binary.

> * To build multi-arch images, you need to have `buildx` installed. Follow these [instructions](https://github.com/docker/buildx#installing) to install buildx.

Reviewer 2: Rename the file as BUILD.md

Reviewer 2: Also, can you push the Git Development workflow into another document named "git-workflow.md"

Author: Done
